server/routers/status-page-router.js

```diff
@@ -84,21 +85,75 @@ router.get("/api/status-page/heartbeat/:slug", cache("1 minutes"), async (reques
             statusPageID
         ]);
 
-        for (let monitorID of monitorIDList) {
-            let list = await R.getAll(`
+        // Get the status page to determine the heartbeat range
+        let statusPage = await R.findOne("status_page", " id = ? ", [ statusPageID ]);
+        let heartbeatBarDays = statusPage.heartbeat_bar_days;
+
+        // Get max beats parameter from query string (for client-side screen width constraints)
+        const maxBeats = Math.min(parseInt(request.query.maxBeats) || 100, 100);
```

Comment:
When I zoom out, the `maxBeats` parameter does not get appended.
This means that there are now 100 beats.
I cannot display 100 beat though, only 68 beats => the beat disapears into the aether.

![image](https://github.com/user-attachments/assets/74a914b4-6b68-4cdb-baa4-b32651f7b5e6)
We cannot cap at 68, because users could extend this in CSS. 100 is still a good limit though imo.

The unfortunate bug where the frontend does not tell the backend the maxBeats needs to be fixed, as otherwise this will cause a lot of confusion.
This appears right around the 1080p border.